dask_jobqueue/oar.py

```diff
@@ -0,0 +1,122 @@
+
+        super(OARCluster, self).__init__(**kwargs)
+
+        header_lines = ['#!/usr/bin/env bash']
+        if self.name is not None:
+            header_lines.append('#OAR -n %s' % self.name)
+        if queue is not None:
+            header_lines.append('#OAR -q %s' % queue)
+        if project is not None:
+            header_lines.append('#OAR --project %s' % project)
+
+        # OAR needs to have the resource on a single line otherwise it is
+        # considered as a "moldable job" (i.e. the scheduler can chose between
+        # multiple sets of resources constraints)
+        resource_spec_list = []
+        if resource_spec is None:
+            # default resource_spec if not specified. Crucial to specify
+            # nodes=1 to make sure the cores allocated are on the same node.
+            resource_spec = '/nodes=1/core=%d' % self.worker_cores
+        resource_spec_list.append(resource_spec)
+        if walltime is not None:
+            resource_spec_list.append('walltime=%s' % walltime)
+
+        full_resource_spec = ','.join(resource_spec_list)
+        header_lines.append('#OAR -l %s' % full_resource_spec)
+        header_lines.extend(['#OAR %s' % arg for arg in job_extra])
+        header_lines.append('JOB_ID=${OAR_JOB_ID}')
+
+        self.job_header = '\n'.join(header_lines)
+
+        logger.debug("Job script: \n %s" % self.job_script())
+
+    def _job_id_from_submit_output(self, out):
+        job_id_regexp = r'OAR_JOB_ID=(?P<job_id>\d+)'
+        match = re.search(job_id_regexp, out)
+        if match is None:
+            raise ValueError('Unexpected output from the submission command')
+        group_dict = match.groupdict()
+        if 'job_id' not in group_dict:
+            raise ValueError('Unexpected output from the submission command')
+        return group_dict['job_id']
+
+    def _submit_job(self, fn):
+        # OAR specificity: the submission script needs to exist on the worker
+        # when the job starts on the worker. This is different from other
+        # schedulers that only need the script on the submission node at
+        # submission time. That means that we can not use the same strategy as
+        # in JobQueueCluster: create a temporary submission file, submit the
+        # script, delete the submission file. In order to reuse the code in
+        # the base JobQueueCluster class, we read from the temporary file and
+        # reconstruct the command line where the script is passed in as a
+        # string (inline script in OAR jargon) rather than as a filename.
+        with open(fn) as f:
+            content_lines = f.readlines()
+
+        oar_lines = [line for line in content_lines if line.startswith('#OAR ')]
+        oarsub_options = [line.replace('#OAR ', '').strip() for line in oar_lines]
+        inline_script_lines = [line for line in content_lines if not line.startswith('#')]
+        inline_script = ''.join(inline_script_lines)
```

Comment:
Won't this cause problem to join on empty string? Joining 'JOB_ID=${OAR_JOB_ID}' with dask-worker command won't lead to something weird?

Reply:
With `.readlines()` you still have the `\n` at the end of the lines.

Reply:
Nevermind, sorry.